src/auth.ts

```diff
@@ -54,6 +76,173 @@ interface GitHubAuthResponse {
 // TODO fix after https://github.com/http-party/node-portfinder/pull/115
 ((portfinder as unknown) as { basePort: number }).basePort = 9005;
 
+/**
+ * Get the global default account. Before multi-auth was implemented
+ * this was the only account.
+ */
+export function getGlobalDefaultAccount(): Account | undefined {
+  const user = configstore.get("user") as User | undefined;
+  const tokens = configstore.get("tokens") as Tokens | undefined;
+
+  // TODO: Is there ever a case where only User or Tokens is defined
+  //       and we want to accept that?
+  if (!user || !tokens) {
+    return undefined;
+  }
+
+  return {
+    user,
+    tokens,
+  };
+}
+
+/**
+ * Get the default account associated with a project directory, or the global default.
+ * @param projectDir the Firebase project directory.
+ */
+export function getProjectDefaultAccount(projectDir?: string | null): Account | undefined {
+  if (!projectDir) {
+    return getGlobalDefaultAccount();
+  }
+
+  const activeAccounts = configstore.get("activeAccounts") || {};
+  const email: string | undefined = activeAccounts[projectDir];
+
+  if (!email) {
+    return getGlobalDefaultAccount();
+  }
+
+  const allAccounts = getAllAccounts();
+  return allAccounts.find((a) => a.user.email === email);
+}
+
+/**
+ * Get all authenticated accounts _besides_ the default account.
+ */
+export function getAdditionalAccounts(): Account[] {
```

Comment:
I'm noticing that all these new functions are exported - is that necessary? Some of these seem like helpers...

Reply:
They're all used in at least one other file.